vector-im/element-android
# FTUE - Server selection errors

## Type of change


## Content

- Uses the TextUri type for url entry (fixes spaces being added when typing)
- Adds missing error states to the `Server Selection` screen instead of when submitting the account details
  - Starts the registration flow at the point of setting the homeserver url instead of waiting until details submission  
- Fixes flashing loading state when selecting a usage option by flattening nested scope launching and only applying loading changes to top level entry points

## Motivation and context

Fixes #5749

## Screenshots / GIFs

|Before|After|
|-|-|
|![before-disabled-reg](https://user-images.githubusercontent.com/1848238/163376485-e021f5c8-efa2-402a-90a6-a273511cdff2.gif)|![after-disabled-reg](https://user-images.githubusercontent.com/1848238/163376507-73c8de5d-f239-4a03-b3ef-e60997b5a5c6.gif)|
|![before-wrong-url](https://user-images.githubusercontent.com/1848238/163376743-5f595978-3abb-43f6-915c-ebd342bbc626.png)|![after-wrong-url](https://user-images.githubusercontent.com/1848238/163376744-094f576f-d4c7-47f4-a132-24ca43aa535d.png)

## Tests


- With the combined register feature flag enabled
- Start the account creation flow
- Edit the server and supply a server with registration disabled or an invalid url
- Notice the error message is now displayed within the entry field

## Tested devices

- OS version(s): 29
